elastic/eui
# [Emotion] Convert EuiHeader and EuiHeaderLogo

## Summary

- Converts `EuiHeader` to Emotion
  - Creates a `euiHeaderVariables()` util (similar to `euiFormVariables()`) for other Emotion components to grab shared variables in the future (mostly height related)
  - Removes all `.euiHeader--` modifier classes
  - Removes several `$euiHeader` Sass variables that are not used by either EUI or Kibana
  - Removes `@euiHeaderDarkTheme` Sass mixin and replaces it with an Emotion util (internal only). Also converts `EuiSelectableTemplateSitewide` CSS that was specific to the header dark theme to be nested within the dark theme
- Removes random unused `.euiHeaderProfile` Sass - appears to only be used in docs and not in actual src code
- Converts `EuiHeaderLogo` to Emotion and moves all related files to a new `header/header_logo` subdirectory
- Does **not** convert all header-related Sass files (https://github.com/elastic/eui/issues/6417) - I'm breaking the work up into multiple smaller portions for easier review/QA

I strongly recommend [following along by commit](https://github.com/elastic/eui/pull/6878/commits).

## QA

- Compare [staging](https://eui.elastic.co/pr_6878/#/layout/header) to [prod](https://elastic.github.io/eui/#/layout/header) and confirm they look the same
- `gh pr checkout 6878` && `yarn storybook`
- EuiHeader: test that the `position` and `theme` props work as expected
- EuiHeaderLogo: test that the `iconType` and `iconTitle` props work as expected
- Go to https://eui.elastic.co/pr_6878/#/layout/flyout and confirm all flyouts do not collide with the header
- Go to https://eui.elastic.co/pr_6878/#/layout/header/elastic-pattern and confirm the collapsible nav flyout does not collide with the header

### General checklist


### Emotion checklist

**Kibana usage**

  ~- [ ] If usage exists, consider converting to a `data` attribute so that consumers still have something to hook into~ - no usages

---

**General**

~- [ ] Checked component playground~ - No playground
~~  > NOTE: Class components wrapped in `withEuiTheme` need to pass `true` as the second argument to its `propUtilityForPlayground` in `src-docs/src/views/{component}/playground.js`~

---

**Unit tests**

- [x] Converted to RTL
~- [ ] Removed any `mount()`ed snapshots in favor of `render()` or a more specific assertion~

---

**Sass/Emotion conversion process**

~- [ ] Ran `yarn compile-scss` to update var/mixin [JSON files](https://github.com/elastic/eui/tree/main/src-docs/src/views/theme/_json)~
~- [ ] Simplified `calc()` to `mathWithUnits` if possible (if mixing different unit types, this may not be possible)~
~- [ ] Added an `@warn` deprecation message within the `global_styling/mixins/{component}.scss` file~
~- [ ] Removed component from `src/components/index.scss`~ Not done with full component conversion yet
- [x] Converted relevant `src/amsterdam/overrides/{component}.scss` styles to baseline Emotion styles

---

**CSS tech debt**

~- [ ] Wrapped all animations or transitions in `euiCanAnimate`~
~- [ ] Used `gap` property to add margin between items if using flex~ - no margins, although possibly padding might be  convertible - will determine in next PR

---

**DOM Cleanup**


---

**Kibana due diligence**

  - ✅ No usages of selectors that aren't the main `.euiHeader` usage
  - ⚠️ [1 usage](https://github.com/elastic/kibana/blob/f7d030fa64ba4ddae103e9c99b74eda004e20b83/x-pack/plugins/security_solution/public/management/components/console/components/side_panel/side_panel_flex_item.tsx#L15)
  - ⚠️ [1 usage](https://github.com/elastic/kibana/blob/a7293f62b59508a4a2fa89fa04c278b70de5901f/packages/core/chrome/core-chrome-browser-internal/src/ui/header/header_logo.tsx#L98)

---

**Extras/nice-to-have**

  - all styles are flattened except for the dark theme styles, which I'm I'm preserving mostly as-is for now with the hope that we eventually can just nuke it w/ the nav redesign

~- [ ] Check for issues in the backlog that could be a quick fix for that component~
~- [ ] Optional component/code cleanup: consider splitting up the component into multiple children if it's overly verbose or difficult to reason about~
~- [ ] Documentation pass:~
